File: codegen/results/v002/qwen3.6_35b-a3b-coding-nvfp4_think/20260419_110709/python/007_beatles_interview/solution.py
import csv
import json
from datetime import date

def main():
    ref_date = date(2025, 7, 1)
    result = []
    
    with open('input/input.csv', 'r') as f:
        reader = csv.DictReader(f)
        for row in reader:
            name_parts = row['Name'].strip().split()
            first_name = name_parts[0]
            last_name = name_parts[-1]
            
            b_parts = row['Birthday'].strip().split('/')
            birthday_str = f"{b_parts[2]}-{b_parts[0].zfill(2)}-{b_parts[1].zfill(2)}"
            
            b_month, b_day, b_year = map(int, b_parts)
            b_date = date(b_year, b_month, b_day)
            age = ref_date.year - b_date.year
            if (ref_date.month, ref_date.day) < (b_date.month, b_date.day):
                age -= 1
                
            relatives = []
            for col, rel in [('Father', 'Father'), ('Mother', 'Mother'), ('Brother', 'Brother'), ('Sister', 'Sister')]:
                val = row[col].strip()
                if val.lower() != 'null':
                    parts = val.split()
                    relatives.append({
                        'FirstName': parts[0],
                        'LastName': parts[-1],
                        'Relationship': rel
                    })
                    
            result.append({
                'FirstName': first_name,
                'LastName': last_name,
                'Birthday': birthday_str,
                'Age': age,
                'Relatives': relatives
            })
            
    print(json.dumps(result, indent=2))

if __name__ == '__main__':
    main()

File: codegen/results/v002/qwen3.6_35b-a3b-coding-nvfp4_think/20260419_110709/go/007_beatles_interview/solution.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"fmt"
	"os"
	"strings"
	"time"
)

type Relative struct {
	FirstName    string `json:"FirstName"`
	LastName     string `json:"LastName"`
	Relationship string `json:"Relationship"`
}

type Person struct {
	FirstName   string     `json:"FirstName"`
	LastName    string     `json:"LastName"`
	Birthday    string     `json:"Birthday"`
	Age         int        `json:"Age"`
	Relatives   []Relative `json:"Relatives"`
}

func main() {
	f, err := os.Open("input/input.csv")
	if err != nil {
		panic(err)
	}
	defer f.Close()

	reader := csv.NewReader(f)
	records, err := reader.ReadAll()
	if err != nil {
		panic(err)
	}

	var people []Person
	refDate := time.Date(2025, 7, 1, 0, 0, 0, 0, time.UTC)

	for i, record := range records {
		if i == 0 {
			continue
		}
		if len(record) < 7 {
			continue
		}

		nameParts := strings.Fields(record[0])
		firstName := nameParts[0]
		lastName := nameParts[len(nameParts)-1]

		bday, err := time.Parse("1/2/2006", strings.TrimSpace(record[1]))
		if err != nil {
			panic(err)
		}
		bdayFormatted := bday.Format("2006-01-02")

		age := refDate.Year() - bday.Year()
		if refDate.Month() < bday.Month() || (refDate.Month() == bday.Month() && refDate.Day() < bday.Day()) {
			age--
		}

		relatives := make([]Relative, 0)
		relPairs := []struct {
			Rel string
			Idx int
		}{
			{"Father", 3},
			{"Mother", 4},
			{"Brother", 5},
			{"Sister", 6},
		}

		for _, rp := range relPairs {
			val := strings.TrimSpace(record[rp.Idx])
			if val == "null" {
				continue
			}
			relParts := strings.Fields(val)
			if len(relParts) >= 2 {
				relatives = append(relatives, Relative{
					FirstName:    relParts[0],
					LastName:     relParts[len(relParts)-1],
					Relationship: rp.Rel,
				})
			}
		}

		people = append(people, Person{
			FirstName:   firstName,
			LastName:    lastName,
			Birthday:    bdayFormatted,
			Age:         age,
			Relatives:   relatives,
		})
	}

	jsonBytes, err := json.MarshalIndent(people, "", "  ")
	if err != nil {
		panic(err)
	}
	fmt.Print(string(jsonBytes))
}